eventuals/task.h

```diff
@@ -238,41 +282,71 @@ struct _TaskFromToWith {
     K_ k_;
     std::tuple<Args_...> args_;
 
-    Callback<
-        Action,
-        std::optional<std::exception_ptr>&&,
-        Args_&&...,
-        std::optional<
-            std::conditional_t<
-                std::is_void_v<From_>,
-                std::monostate,
-                From_>>&&,
-        std::unique_ptr<void, Callback<void*>>&,
-        Interrupt&,
+    // The 'dispatch_' is a `std::variant` because its either a function that
+    // creates an eventual or the value from 'Task::Success' that should be
+    // passed on to the continuation.
+    std::variant<
         std::conditional_t<
             std::is_void_v<To_>,
-            Callback<>&&,
-            Callback<To_>&&>,
-        Callback<std::exception_ptr>&&,
-        Callback<>&&>
+            std::monostate,
+            std::conditional_t<
+                !std::is_reference_v<To_>,
+                To_,
+                std::reference_wrapper<
+                    std::remove_reference_t<To_>>>>,
+        Dispatch_>
         dispatch_;
 
     std::unique_ptr<void, Callback<void*>> e_;
     Interrupt* interrupt_ = nullptr;
   };
 
-  template <typename From_, typename To_, typename... Args_>
+  template <
+      typename From_,
+      typename To_,
+      typename... Args_>
   struct Composable {
+    using Dispatch_ =
+        Callback<
+            Action,
+            std::optional<std::exception_ptr>&&,
+            Args_&&...,
+            // Can't have a 'void' argument type
+            // so we are using 'std::monostate'.
+            std::optional<
+                std::conditional_t<
+                    std::is_void_v<From_>,
+                    std::monostate,
+                    From_>>&&,
+            std::unique_ptr<void, Callback<void*>>&,
+            Interrupt&,
+            std::conditional_t<
+                std::is_void_v<To_>,
+                Callback<>&&,
+                Callback<To_>&&>,
+            Callback<std::exception_ptr>&&,
+            Callback<>&&>;
+
     template <typename>
     using ValueFrom = To_;
 
+    Composable(
+        std::conditional_t<
+            std::is_void_v<To_>,
+            std::monostate,
+            std::conditional_t<
+                !std::is_reference_v<To_>,
+                To_,
+                std::reference_wrapper<
+                    std::remove_reference_t<To_>>>> value)
+      : dispatch_(std::move(value)) {}
```

Comment:
```suggestion
    Composable(MonostateIfVoidOrReferenceWrapperOr<To_> value)
      : value_or_dispatch_(std::move(value)) {}
```

And let's rename `dispatch_` to `value_or_dispatch_` to make it even more clear.